### ISSUE TITLE
<AutocompleteInput> optionText is a React element

### ISSUE THREAD

**Issue by wmwart** [marmelab/react-admin#4092]
**What you were expecting:**
Expecting a custom item as a drop down list item

**What happened instead:**
In version 2, the component supported the suggestionComponent parameter. Now there is nothing in the documentation about this, however, in the [Properties table,](https://marmelab.com/react-admin/Inputs.html#autocompleteinput) the description for the matchSuggestion parameter indicates:
`Required if optionText is a React element. Function returning a boolean indicating whether a choice matches the filter.` 
I concluded that AutocompleteInput still supports the React component.
I wrote my code and got an error:

> Warning: Failed prop type: Invalid prop `inputValue` of type `object` supplied to `Downshift`, expected `string`.
```
const optionRenderer = choice => {

return (
        <span>{choice.name}-{choice.description}</span>
    )
};

const optionMatch = (filter, choice) => {

    return true
};

const GasMeterEdit = props => {
    const controllerProps = useEditController(props);
    const isDeleted = controllerProps.record && controllerProps.record.deleted;

    return (
        <EditView 
            title={<TitleForRecord source="name" {...props}/>}
            {...props}
            {...controllerProps}
        >
            <SimpleForm toolbar={<GasMeterToolbar />}>
                <ReferenceInput source="controller.id" reference="Controller" label="resources.GasMeter.fields.controller" allowEmpty emptyText="ra.message.no" options={{disabled: isDeleted}} >
                    <AutocompleteInput optionText={optionRenderer} matchSuggestion={optionMatch}/>
                </ReferenceInput>           
            </SimpleForm>
        </EditView>
    );
};
```
**Steps to reproduce:**
1) set optionText and matchSuggestion properties for AutocompleteInput
2) optionRenderer return React element

**Other information:**
```
index.js:1375 Warning: Failed prop type: Invalid prop `inputValue` of type `object` supplied to `Downshift`, expected `string`.
    in Downshift (created by AutocompleteInput)
    in AutocompleteInput (at GasMeterEdit.js:81)
    in ReferenceInputView (created by ReferenceInput)
    in ReferenceInput (at GasMeterEdit.js:80)
    in div (created by FormInput)
    in FormInput (created by SimpleFormView)
    in div (created by ForwardRef(CardContent))
    in ForwardRef(CardContent) (created by WithStyles(ForwardRef(CardContent)))
    in WithStyles(ForwardRef(CardContent)) (created by CardContentInner)
    in CardContentInner (created by SimpleFormView)
    in form (created by SimpleFormView)
    in SimpleFormView (created by ReactFinalForm)
    in ReactFinalForm (created by SimpleForm)
    in SimpleForm (at GasMeterEdit.js:73)
    in div (created by ForwardRef(Paper))
    in ForwardRef(Paper) (created by WithStyles(ForwardRef(Paper)))
    in WithStyles(ForwardRef(Paper)) (created by ForwardRef(Card))
    in ForwardRef(Card) (created by WithStyles(ForwardRef(Card)))
    in WithStyles(ForwardRef(Card)) (created by EditView)
    in div (created by EditView)
    in div (created by EditView)
    in EditView (at GasMeterEdit.js:68)
    in GasMeterEdit (created by WithPermissions)
    in WithPermissions (created by Context.Consumer)
    in Route (created by ResourceRoutes)
    in Switch (created by ResourceRoutes)
    in ResourceRoutes (created by Resource)
    in Resource (at App.js:57)
    in Route (created by RoutesWithLayout)
    in Switch (created by RoutesWithLayout)
    in RoutesWithLayout (created by Context.Consumer)
    in div (created by Layout)
    in main (created by Layout)
    in div (created by Layout)
    in div (created by Layout)
    in Layout (created by WithStyles(Layout))
    in WithStyles(Layout) (created by Context.Consumer)
    in withRouter(WithStyles(Layout)) (created by ConnectFunction)
    in ConnectFunction (created by LayoutWithTheme)
    in ThemeProvider (created by LayoutWithTheme)
    in LayoutWithTheme (at Layout/index.js:9)
    in MyLayout (created by Context.Consumer)
    in Route (created by CoreAdminRouter)
    in Switch (created by CoreAdminRouter)
    in div (created by CoreAdminRouter)
    in CoreAdminRouter (created by Context.Consumer)
    in Route (created by AdminUI)
    in Switch (created by AdminUI)
    in AdminUI (created by Admin)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (created by Context.Consumer)
    in ConnectedRouterWithContext (created by ConnectFunction)
    in ConnectFunction (created by AdminContext)
    in TranslationProvider (created by AdminContext)
    in Provider (created by AdminContext)
    in AdminContext (created by Admin)
    in Admin (at App.js:50)
    in App (at src/index.js:7)
```

**Environment**
If it’s not difficult, provide a code example of how this should work.

* React-admin version: 3.0.2

**Comment by fzaninotto** [comment]
Confirmed and reproduced, thanks

**Comment by JulienMattiussi** [comment]
This have been solved by #4367

You now have to add also an `inputText` prop to your AutocompleteInput when the `optionText` is a Component

https://marmelab.com/react-admin/doc/3.2/Inputs.html#autocompleteinput
